mkosi/backend.py

```diff
@@ -948,3 +949,25 @@ def complete_step(cls, text: str, text2: Optional[str] = None) -> Iterator[List[
 
         if text2 is not None:
             cls.print_step(text2.format(*args))
+
+
+def chown_to_running_user(path: PathString) -> None:
+    uid = int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or str(os.getuid()))
+    user = pwd.getpwuid(uid).pw_name
+    gid = pwd.getpwuid(uid).pw_gid
+
+    with MkosiPrinter.complete_step(
+        f"Changing ownership of output file {path} to user {user}…",
+        f"Changed ownership of {path}",
+    ):
+        os.chown(path, uid, gid)
+
+
+def mkdir_chown_current_user(path: Path, skip_chown: bool = False, mode: int = 0o777) -> None:
+    if os.path.isdir(path):
+        return
+
+    path.mkdir(mode=mode)
+
+    if not skip_chown:
+        chown_to_running_user(path)
```

Comment:
This gives the function a very weird semantic, because it runs counter to the name. How about just naming the function `mkdir`, plumbing through all the options of `Path.mkdir` and giving it additional keyword arguments `owner=None` and `group=None` that will be passed to a `chown` at the end?

Reply:
Lennart (I think?) suggested scoping `SUDO_UID` and `PKEXEC_UID` into a specific function, in this case it would be `chown_to_running_user()`, so having `owner=` and `group=` argument would go against that. I used @labichn's code to create and chown missing directories recursively and added `if skip_chown: continue` to skip the call to `chown_to_running_user()`. This makes the code flow a bit better.

Reply:
I can live with the structure of the code, but the name just makes what the function does non-obvious, because the name doesn't really imply that the `chown` would be optional, but I'm a bit hard-pressed to come up with a better name at the moment.